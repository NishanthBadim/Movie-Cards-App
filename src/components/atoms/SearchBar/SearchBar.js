import React from 'react';
import '../../../index.css'

  
const SearchBar =(props) => {
    return(
    <div className="form-group has-search">
        <span className="fa fa-search form-control-feedback"></span>
        <input style={{display:'inline-block'}}  type="text" className="form-control" size="40" value={props.value} onChange={props.changed} placeholder="Search by Movie Name..." /> 
        <p className={props.result==='No Results Found' ? 'text-results' : ''} style={{display:'inline-block',fontSize:'18px',color:'gray'}}>{props.result}</p>
    </div>
    );
}
export default SearchBar;