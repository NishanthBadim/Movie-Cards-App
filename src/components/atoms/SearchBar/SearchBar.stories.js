import React from 'react'
import { withKnobs, text } from '@storybook/addon-knobs/react'
import { action } from '@storybook/addon-actions'
import './../../../index.css'

import SearchBar from './SearchBar'

export default {
    component: SearchBar,
    title: "SearchBar",
    decorators: [withKnobs]
}
export const Default = () => {
    return (
        <div className="form-group has-search">
            <span className="fa fa-search form-control-feedback"></span>
            <SearchBar value={text("value", "")} result={text("results", "")}
                changed={action("Entered Input")} />
        </div>
    )
}
export const MultipleResults = () => {
    return (
        <SearchBar value={text("value", "")} result={text("results", " 2 Results Found")}
            changed={action("Entered Input")} />
    )
}

export const SingleResult = () => {
    return (
        <SearchBar value={text("value", "")} result={text("results", " 1 Result Found")}
            changed={action("Entered Input")} />
    )
}

export const NoResults = () => {
    return (
        <SearchBar value={text("value", "")} result={text("results", "No Results Found")}
            changed={action("Entered Input")} />
    )
}