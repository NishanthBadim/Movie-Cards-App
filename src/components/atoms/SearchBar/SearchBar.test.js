import React from 'react';
import {create} from 'react-test-renderer'
import "@testing-library/jest-dom/extend-expect";
import SearchBar from './SearchBar'
import {render, fireEvent} from '@testing-library/react'

test('Testing the Search Bar input', () => {
    const handleChange = jest.fn()
    const { container } = render(<input type="text" onChange={handleChange} />)
    const input = container.firstChild
    fireEvent.change(input, { target: { value: 'a' } })
    expect(handleChange).toHaveBeenCalledTimes(1)
    expect(input.value).toBe('a')
})

test("Match the snapshot - SearchBar",() => {
    const input=create(<SearchBar />);
    expect(input.toJSON()).toMatchSnapshot();
})