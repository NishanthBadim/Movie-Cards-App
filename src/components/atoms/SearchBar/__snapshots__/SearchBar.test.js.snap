// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot - SearchBar 1`] = `
<div
  className="form-group has-search"
>
  <span
    className="fa fa-search form-control-feedback"
  />
  <input
    className="form-control"
    placeholder="Search by Movie Name..."
    size="40"
    style={
      Object {
        "display": "inline-block",
      }
    }
    type="text"
  />
  <p
    className=""
    style={
      Object {
        "color": "gray",
        "display": "inline-block",
        "fontSize": "18px",
      }
    }
  />
</div>
`;
